src/resolve-dependency.ts

```diff
@@ -259,8 +259,18 @@ async function resolveRemappings(
 ): Promise<void> {
   if (job.conditions?.includes('browser')) {
     const { browser: pkgBrowser } = pkgCfg;
+    if (!pkgBrowser) {
+      return;
+    }
```

Comment:
Actually now that I think about it, isn't there a case where browser can be a string?

<img width="847" alt="image" src="https://github.com/vercel/nft/assets/229881/e9f0f07c-9369-4423-b351-5828748702dc">

We should add a test for this case as well
https://gist.github.com/defunctzombie/4339901/49493836fb873ddaa4b8a7aa0ef2352119f69211

Reply:
It can be, but I saw [this test](https://github.com/vercel/nft/blob/main/test/unit/browser-remappings-malformed/node_modules/pkg/package.json) and now I think that maybe we should update its implementation (and rename). It's ensuring that the field specified in the browser field is not in the output, but it's not imported anywhere in the other modules so I think it's a false positive (as you pointed out when we were pairing).

Reply:
@jeffsee55 That is very strange. 

Maybe talk to @dvoytenko to see if that is a mistake since I would expect to see [import-main-browser.js](https://github.com/vercel/nft/blob/main/test/unit/browser-remappings-malformed/node_modules/pkg/subdir/import-main-browser.js) in the [output.js](https://github.com/vercel/nft/blob/main/test/unit/browser-remappings-malformed/output.js).

Also, that test seems very oddly written since `pkg` is its own package boundary and `pkg/subdir` is another package boundary, but the `browser` field is referencing files from one to the other. I don't think thats common (or even allowed?)

Reply:
Thanks @styfle , will follow up with Dima.

For posterity, here are my thoughts on the current state of these tests, it seems like the `output.js` should expect only files that are traceable from `nft`, but in the tests the browser-specific files aren't imported or required by the paths the relate to the input entrypoint. So in my test I [required](https://github.com/vercel/nft/pull/427/files#diff-b636571d2e3ec4384710db3b1735419c20622cc2efb06b818af0475347b15f37R2) it specifically. So it seemed like the other tests aren't asserting the right behavior because those browser files just aren't part of the dependency graph.

Reply:
I think it could be a string, as in "main export file for browser". TBH, I haven't seen those in the wild and not what I've been trying to support in one of my previous PRs. But I think it's a use case in its own right and probably should be supported too. It's a relatively under-defined and legacy area - most of folk moved away from `browser`, but I found the best way to debug them is by trying out to build examples using `esbuild` - it understands `browser` correctly.